src/Libraries/Microsoft.AspNetCore.HeaderParsing/Microsoft.AspNetCore.HeaderParsing.csproj

```diff
@@ -8,11 +8,15 @@
 
   <PropertyGroup>
     <TargetFrameworks>$(NetCoreTargetFrameworks)</TargetFrameworks>
-    <EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>
+    <EnableConfigurationBindingGenerator Condition="'$(TargetFramework)' == 'net9.0'">true</EnableConfigurationBindingGenerator>
     <UseLoggingGenerator>true</UseLoggingGenerator>
     <UseMetricsGenerator>true</UseMetricsGenerator>
     <InjectExperimentalAttributeOnLegacy>true</InjectExperimentalAttributeOnLegacy>
     <InjectGetOrAddOnLegacy>true</InjectGetOrAddOnLegacy>
+
+    <!-- using the ConfigurationBinder source generator is blocked by https://github.com/dotnet/runtime/issues/94547 -->
+    <IsAotCompatible Condition="'$(TargetFramework)' != 'net9.0'">false</IsAotCompatible>
```

Comment:
Can you make this an "IsCompatible" check? So when we add net10 it works correctly.

Reply:
`$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net9.0'))`

Reply:
This comment hasn't been resolved.